test/broadcast.jl

```diff
@@ -355,3 +355,7 @@ StrangeType18623(x,y) = (x,y)
 
 # 19419
 @test @inferred(broadcast(round, Int, [1])) == [1]
+
+let f(A, n) = broadcast(x -> +(x, n), A)
+    @test @inferred(f([1.0], 1)) == [2.0]
+end
```

Comment:
Perhaps point to the discourse thread, and include an analog of the direct test of the promotion mechanism from that thread? For example,
```julia
# Test issue described in https://discourse.julialang.org/t/towards-broadcast-over-combinations-of-sparse-matrices-and-scalars/910
let
    f(A, n) = broadcast(x -> +(x, n), A)
    @test @inferred(f([1.0], 1)) == [2.0]
    foo() = (a = 1.0; Base.Broadcast._broadcast_type(x -> x + a, 1.0))
    @test @inferred(foo()) == Float64
end
```